### ISSUE TITLE
Scripts about fully unsupervised learning on training and evaluation?

### ISSUE THREAD

**Issue by JizeCao** [ByungKwanLee/Causal-Unsupervised-Segmentation#5]
According to the paper, the step 2 should perform self-supervised learning with contrastive learning objective. However, in the script _train_front_door_tr.py_ and _fine_tuning_tr.py_ , the dataset ground-truth labels are used to train the linear probe. And that linear probe is also used to generate predictions during the inference . 

I understand that these scripts are used to generate Table 2 result, which uses a linear probe to compare against other baselines.

Is there any script that performs fully unsupervised learning training & evaluation in this repo ? That would be really helpful if such scripts exist for us to evaluate this method's performance in a fully unsupervised learning manner without linear probing. :)

**Comment by ByungKwanLee** [comment]
You can find the following line in `train_front_door_tr.py` [[Link](https://github.com/ByungKwanLee/Causal-Unsupervised-Segmentation/blob/48e6459d4f9f57ba732084d230a13c0ada70c6df/train_front_door_tr.py#L210C1-L213C67)]
```python
# Bank and EMA
cluster.bank_init()
ema_init(segment.head, segment.head_ema)
ema_init(segment.projection_head, segment.projection_head_ema
```

Here, the funcion of `ema_init` is defined in `modules/segment_module.py` [[Link](https://github.com/ByungKwanLee/Causal-Unsupervised-Segmentation/blob/48e6459d4f9f57ba732084d230a13c0ada70c6df/modules/segment_module.py#L342)]

```python
def ema_init(x, x_ema):
    for param, param_ema in zip(x.parameters(), x_ema.parameters()): param_ema.data = param.data; param_ema.requires_grad = False
```

**Comment by thingumajig** [comment]
I join the question. It is not entirely clear how to run fully self-supervised learning on an arbitrary set of images.

**Comment by ChristophReich1996** [comment]
Same here! I have doubts that the approach is fully unsupervised, as I see no part in the code where the compute graph is cut (tensor.detach()) such that no gradient information can leak from the linear probing. Am I missing something here? @ByungKwanLee
